src/modules/cron/task.c

```diff
@@ -112,11 +110,13 @@ void cron_task_on_state_change (cron_task_t *t, cron_task_state_f cb)
     t->state_cb = cb;
 }
 
-static bool cron_task_completed (cron_task_t *t)
+static bool cron_task_finished (cron_task_t *t)
 {
     if (t->rexec_failed)
         return true;
-    if (t->exited && t->stderr_closed && t->stdout_closed)
+    if (t->exec_failed)
+        return true;
+    if (t->completed)
         return true;
```

Comment:
I'm not sure I understand the gratuitous change of the word `completed` to `finished` in the cron module. Especially when you look at `cron_task_finished` which returns true when the completed flag is set... Is this because it was confusing that a task could be completed but not actually started because rexec failed?

Reply:
The main reason for the change was that the original cron task code used the word "completed" to mean that a task was no longer running, and that included if the task failed with an error (like rexec failed).  This is in contrast to the subprocess API, where "completed" means exited & io flushed.

I just changed it to differentiate it.  It is purely my style choice, could revert it.

Reply:
Ah, ok. Since the task code is a big user of the updated subprocess interface, and you made the switch, it makes sense that you first made the code comfortable to work with. In the future I would say something like that would be a separate PR, but it makes sense here.

Reply:
I did ponder just splitting off the last 4-5 commits into a separate PR, since they are really just   cleanup of cron task.  I'll remove them the next time I push and we'll save it for another PR.